File: src/lib/api/copilot.ts
import { API_BASE_URL } from "./client";

export interface CopilotHistoryMessage {
  role: "physician" | "assistant";
  content: string;
}

export interface CopilotChatPayload {
  encounter_id: string;
  patient_record_number?: string;
  message: string;
  conversation_history: CopilotHistoryMessage[];
  soap_note_context?: unknown;
}

export interface CopilotToolCallResult {
  tool: string;
  complaint_type?: string;
  missing?: string[];
  present?: string[];
  completeness_score?: number;
  flags_detected?: string[];
  flags_missing?: string[];
}

export interface CopilotChatResponse {
  assistant_message: string;
  /** backward compat — old API used this field */
  response?: string;
  tool_calls?: CopilotToolCallResult[];
  suggested_follow_ups?: string[];
  sources?: string[];
}

export async function sendCopilotMessage(payload: CopilotChatPayload): Promise<CopilotChatResponse> {
  const res = await fetch(`${API_BASE_URL}/api/v1/copilot/chat`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload)
  });
  if (!res.ok) throw new Error(`Copilot unavailable: ${res.status}`);
  return res.json();
}

File: api/app/routers/copilot.py
from fastapi import APIRouter, HTTPException

from app.schemas.encounters import CopilotChatRequest
from app.services.copilot_service import CopilotService

router = APIRouter()
_copilot = CopilotService()


@router.post("/chat")
def copilot_chat(body: CopilotChatRequest):
    try:
        result = _copilot.chat(
            message=body.message,
            encounter_id=body.encounter_id,
            patient_record_number=body.patient_record_number,
            conversation_history=[m.model_dump() for m in (body.conversation_history or [])],
            soap_note_context=body.soap_note_context,
        )
    except Exception as exc:
        raise HTTPException(status_code=500, detail=str(exc))

    return {
        "assistant_message": result["assistant_message"],
        "sources": result["sources"],
        "suggested_follow_ups": result["suggested_follow_ups"],
        "tool_calls": result.get("tool_calls", []),
    }
